### PR TITLE
Get 'all' nodes directly from CouchDB

### PR DESCRIPTION
**Pull request by skazi0**
**Why is this change necessary?**
Listing all nodes can take a long time on big installations. Our SCALE lab tests show ~15s for listing 230+ nodes. Same applies to dashboard load times.

**How does it address the issue?**
This improves performance of dashboard and crowbarctl node commands by ~30% on big installations. In our SCALE lab time for listing nodes went down from ~15s to ~9.5s.
Chef API and Solr search is skipped. Main benefit comes from avoiding multiple JSON (de)serializations.

**Is there additional information worth sharing like links to a Trello
card, bug references, testing advice or dependencies to other pull
requests?**
https://trello.com/c/kQ1eLgYY/122-evaluate-performance-of-admin-node-on-scale-cloud

### CODE REVIEW

**Comment by skazi0** [comment]
[Evaluate performance of admin node on scale cloud](https://trello.com/c/kQ1eLgYY/122-evaluate-performance-of-admin-node-on-scale-cloud)

File: crowbar_framework/app/models/chef_object.rb [hunk 1]
```diff
@@ -30,6 +30,16 @@ def self.query_chef
...
   end
 
+  def self.fetch_nodes_from_cdb
+    # get all nodes directly from CouchDB, skip Chef API and Solr search
+    nodes = Chef::Node.cdb_list(true)
+    return [nodes, 0, nodes.count]
+  rescue Errno::ECONNREFUSED => e
+    raise Crowbar::Error::ChefOffline
+  rescue StandardError
+    return Chef::Node.new
```
[the remark this answers is not in the document]
Comment (skazi0) [reply crowbar_framework/app/models/chef_object.rb]: You're right. I changed it to return empty result in proper format.

**Comment by skazi0** [comment]
@rhafer my tests show that we're throwing around ~40MB of JSON for this SCALE cloud... a couple of re-serializing rounds takes a lot of time.

**Comment by Itxaka** [comment]
This looks good but I wonder why would it be faster to fetch them directly rather than go through solr. Isnt solr supposed to cache searches to make them fast(-er?, ish?)?

Maybe solr is wrongly configured?

**Comment by skazi0** [comment]
@Itxaka re solr: from Chef code it looks like it is used only for indexing and not for storing values. Result of search is a set of document IDs which are then fetched from CouchDB. If we want to look for config improvements then CouchDB would be a better place to start I guess...

**Comment by Itxaka** [comment]
mmmh, interesting, I expected solr to be some kind of elasticsearch(lucene). Thanks @skazi0 for the hint!

**Comment by skazi0** [comment]
@Itxaka maybe it could be used that way but it looks like in this case it is not :)

**Pull request merged**